src/lib.rs

```diff
@@ -794,6 +795,28 @@ pub trait Itertools : Iterator {
         Combinations::new(self)
     }
 
+    /// Return an iterator adaptor that iterates over the `n`-length combinations of
+    /// the elements from an iterator.
+    ///
+    /// Iterator element type is `Vec<Self::Item, Self::Item>`.
```

Comment:
Just `Vec<Self::Item>`. We should stress that the iterator produces a vec per iteration, and clones the iterator elements.